### DISCUSSION TITLE
Fuego Extensions

### DISCUSSION THREAD

**Discussion by EwenQuim** [go-fuego/fuego#118] [Ideas]
I don't like adding a lot of dependencies to Fuego, but I have the gut feeling that Fuego extensions like Markdown #116 shouldn't be in the default **github.com/go-fuego/fuego** module. It creates noise in the dependencies tree and make binaries heavier. It might also be very useless for users that use Fuego only for some features.

But we'll probably add a lot of functionalities and helpers like this one through the development. I want to keep this project as a monorepo because I like to centralize this, so here's my proposition : 

- Create a module `/extra/markdown`
- The user will have to `go get github.com/go-fuego/fuego/extra/markdown` and the use `markdown.RenderSafeHTML` or `markdown.RenderDangerousHTML` functions

This is the same behaviour as middlewares, except it's for utility functions and more.

What do you think about it ?

**Comment by dylanhitt** [comment]
Yes, let's go with that. 

You're definitely the person to implement this as it will show the pattern moving into the future. I think this will also help with less noise around the core development of fuego.

**Comment by dylanhitt** [comment]
My only initial thoughts is since we're planning on keeping this a monorepo does it change the CI at all? Probably not initially we can just continue doing the catch all rules that we're using right now. Just something to think about, not a show stopper.

**Comment by EwenQuim** [comment]
I think we'll move the Security package down to an extension. It is currently incomplete and needs some deep work that I cannot provide right now
